datasets/noaa_nclimgrid/tests/test_dataset.py

```diff
@@ -0,0 +1,37 @@
+import logging
+from pathlib import Path
+
+from pctasks.cli.cli import setup_logging, setup_logging_for_module
+from pctasks.dev.test_utils import run_process_items_workflow
+
+HERE = Path(__file__).parent
+DATASET_PATH = HERE / ".." / "dataset.yaml"
+
+
+# Change the COG_CONTAINER in noaa_nclimgrid.py to "devstoreaccount1" to avoid
+# uploading COGs to the nclimgridwesteurope storage account
+def test_daily():
+    run_process_items_workflow(
+        DATASET_PATH,
+        collection_id="noaa-nclimgrid-daily",
+        args={
+            "registry": "localhost:5001",
+        },
+        chunks_limit=1,
+    )
+
+
+def test_monthly():
+    pass
+    # The monthly data is one asset chunkline for all data. This is onerous to
+    # run for testing. Note that all monthly data was ingested into PC Test,
+    # which can serve as a proxy for an end-to-end integration test for the
+    # monthly collection.
+
+
+if __name__ == "__main__":
+    setup_logging(logging.DEBUG)
+    setup_logging_for_module("__main__", logging.DEBUG)
+    test_daily()
+    print("All tests passed")
+    exit(0)
```

Comment:
This module _could_ be discoverable by `pytest`, making this `if __name__ == "__main__":` block unnecessary.

If we used pytest, we could also use a `--runslow` option (e.g. https://github.com/stactools-packages/modis/blob/524bfd26bb6d571336e50a226e853b97287529e3/tests/conftest.py) to skip onerous tests by default, but allow them to be run via the command line with a simple flag.

Reply:
>This module could be discoverable by pytest, making this if __name__ == "__main__": block unnecessary.

Agreed. I'm following the pattern of the existing datasets. But you can still run pytest against the file, so no harm. 

I'll look into the --runslow option. I've seen that but have yet to implement it.

Reply:
Done.